### PR TITLE
Add new chart: Thanos

### PR DESCRIPTION
**Pull request by juan131**
Signed-off-by: juan131 <juan@bitnami.com>

**Description of the change**

This PR adds a new chart to the catalog: **Thanos**

[Thanos](https://thanos.io/) is a highly available metrics system that can be added on top of existing Prometheus deployments, providing a global query view across all Prometheus installations.

Please read carefully the **README.md** and provide feedback about anything you think that doesn't make sense.

---

IMPORTANT: To test this chart, I highly recommend following the steps described in the section **"Integrate Thanos with Prometheus and Alertmanager"**

### CODE REVIEW

File: bitnami/thanos/README.md [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+## Introduction
+
+This chart bootstraps a [Thanos](https://github.com/bitnami/bitnami-docker-thanos) deployment on a [Kubernetes](http://kubernetes.io) cluster using the [Helm](https://helm.sh) package manager.
+
+Bitnami charts can be used with [Kubeapps](https://kubeapps.com/) for deployment and management of Helm Charts in clusters.
+
+## Prerequisites
+
+- Kubernetes 1.12+
+- Helm 2.11+ or Helm 3.0-beta3+
```
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/thanos/README.md]: We should also start thinking about putting Helm 3.0 instead of beta3

File: bitnami/thanos/values.yaml [hunk 2]
```diff
@@ -0,0 +1,944 @@
...
+    #   memory: 128Mi
+    requests: {}
+    #   cpu: 100m
+    #   memory: 128Mi
+  
+  ## Thanos Querier pods' liveness and readiness probes
+  ## ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#container-probes
+  ##
+  livenessProbe:
+    httpGet:
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/thanos/values.yaml]: It simply gives more flexibility to the user to customize the probes.

File: bitnami/thanos/templates/bucketweb/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+            - --timeout={{ .Values.bucketweb.timeout }}
+            {{- end }}
+            {{- range $key, $value := .Values.bucketweb.extraFlags }}
+            - --{{ $key }}{{ if $value }}={{ $value }}{{ end }}
+            {{- end }}
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          {{- if .Values.bucketweb.resources }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/thanos/templates/bucketweb/deployment.yaml]: I've been checking other existing Thanos charts and they define readiness/liveness probes for every component but Bucket Web 🤷‍♂ 
It seems it's he only component with no `/-/healthy` and `/-/ready` endpoints.

File: bitnami/thanos/templates/compactor/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+            - name: objstore-config
+              mountPath: /conf/objstore.yml
+              subPath: objstore.yml
+            - name: data
+              mountPath: /data
+      volumes:
+        - name: objstore-config
+          configMap:
+            name: {{ include "thanos.objstoreConfigmapName" . }}
+  {{- if and .Values.compactor.persistence.enabled .Values.compactor.persistence.existingClaim }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/thanos/templates/compactor/statefulset.yaml]: Compactor has an unique replica

File: bitnami/thanos/templates/querier/hpa.yaml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    app.kubernetes.io/component: querier
+spec:
+  scaleTargetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "thanos.fullname" . }}-querier
+  minReplicas: {{ .Values.querier.autoscaling.minReplicas }}
+  maxReplicas: {{ .Values.querier.autoscaling.maxReplicas }}
+  metrics:
+    {{- with .Values.querier.autoscaling.targetCPUUtilizationPercentage }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/thanos/templates/querier/hpa.yaml]: 👍

File: bitnami/thanos/templates/querier/sd-configmap.yaml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{{- if (include "thanos.querier.createSDConfigmap" .) }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ include "thanos.fullname" . }}-querier-sd-configmap
+  labels: {{- include "thanos.labels" . | nindent 4 }}
+    app.kubernetes.io/component: querier
+data:
+{{- if (.Files.Glob "files/conf/servicediscovery.yml") }}
+{{ (.Files.Glob "files/conf/servicediscovery.yml").AsConfig | indent 2 }}
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/thanos/templates/querier/sd-configmap.yaml]: I don't know.. Whatever you prefer...

File: bitnami/thanos/values.yaml [hunk 7]
```diff
@@ -0,0 +1,944 @@
...
+      path: /-/healthy
+      port: http
+    initialDelaySeconds: 30
+    timeoutSeconds: 30
+    # periodSeconds: 10
```
[the remark this answers is not in the document]
Comment (juan131) [reply bitnami/thanos/values.yaml]: Using default K8s values

File: bitnami/thanos/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/thanos/values.yaml]: Then maybe it makes sense to make it a template

File: bitnami/thanos/templates/bucketweb/deployment.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/thanos/templates/bucketweb/deployment.yaml]: Maybe we can add a simple port check

File: bitnami/thanos/templates/compactor/statefulset.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/thanos/templates/compactor/statefulset.yaml]: Then maybe it makes sense to just be a deployment with a service.

File: bitnami/thanos/templates/querier/sd-configmap.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (javsalgar) [reply bitnami/thanos/templates/querier/sd-configmap.yaml]: I would not include it

File: bitnami/thanos/README.md [hunk 8]
```diff
@@ -0,0 +1,559 @@
...
+helm install --name thanos --namespace thanos bitnami/thanos
+```
+
+For Helm 3:
+
+```bash
+helm repo add bitnami https://charts.bitnami.com/bitnami
+kubectl create namespace thanos
+helm install thanos --namespace thanos bitnami/thanos
+```
```
Comment (carrodher) [inline bitnami/thanos/README.md]: Please use the common way to maintain the same structure than other charts:

> ## TL;DR;
> 
> ```console
> $ helm repo add bitnami https://charts.bitnami.com/bitnami
> $ helm install my-release bitnami/thanos
> ```

please, note the `helm repo add` URL without ending in `/bitnami` and just the Helm 3 approach.

File: bitnami/thanos/README.md [hunk 9]
```diff
@@ -0,0 +1,559 @@
...
+helm install --name my-release bitnami/thanos --namespace my-namespace bitnami/thanos
+```
+
+For Helm 3:
+
+```bash
+helm repo add bitnami https://charts.bitnami.com/bitnami
+kubectl create namespace my-namespace
+helm install my-release --namespace my-namespace bitnami/thanos
+```
```
Comment (carrodher) [inline bitnami/thanos/README.md]: Same comment

File: bitnami/thanos/README.md [hunk 10]
```diff
@@ -0,0 +1,559 @@
...
+|----------------------------|----------------------------------------------------------------|-------------------------------------------|
+| `minio.enabled`            | Enable/disable Minio chart installation                        | `false`                                   |
+| `minio.accessKey.password` | MinIO Access Key                                               | _random 10 character alphanumeric string_ |
+| `minio.secretKey.password` | MinIO Secret Key                                               | _random 40 character alphanumeric string_ |
+| `minio.defaultBuckets`     | Comma, semi-colon or space separated list of buckets to create | `nil`                                     |
+
+Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example using Helm 2,
+
+```bash
+helm install --name my-release \
```
Comment (carrodher) [inline bitnami/thanos/README.md]:
```suggestion
helm install my-release \
```

File: bitnami/thanos/README.md [hunk 11]
```diff
@@ -0,0 +1,559 @@
...
+  --set querier.replicaCount=2 \
+    bitnami/thanos
+```
+
+The above command install Thanos chart with 2 Thanos Querier replicas.
+
+Alternatively, a YAML file that specifies the values for the parameters can be provided while installing the chart. For example using Helm 2,
+
+```bash
+helm install --name my-release -f values.yaml bitnami/thanos
```
Comment (carrodher) [inline bitnami/thanos/README.md]:
```suggestion
helm install -f values.yaml bitnami/thanos
```

File: bitnami/thanos/README.md [hunk 12]
```diff
@@ -0,0 +1,559 @@
...
+kubectl create namespace monitoring
+helm install prometheus-operator \
+    --set prometheus.thanos.create=true \
+    --namespace monitoring \
+    bitnami/prometheus-operator
+helm install thanos \
+    --values values.yaml \
+    --namespace monitoring \
+    bitnami/thanos
+```
```
Comment (carrodher) [inline bitnami/thanos/README.md]: Same
